gitpod-io/gitpod
# fixes for task terminals

#### What it does

- fix #2086:
  - align shell with current IDE expectations
  - remove unnecessary line delimiters in task commands
- fix #2208: ensure task status order
- always run task terminals for regular workspaces to avoid closing terminals on workspace restart
- redirect from the prebuild to a workspace on the start page only when the prebuild is stopped to avoid opening a workspace without the prebuild
- don't hang the create page if the prebuild is done but not available
- resolved the read/write deadlock for task terminals
- fix #2265: ensure that headless tasks done is reported always

#### How to test

- Start a regular workspace and check task output: http://akosyakov-verbose-output-from-gitpod-2086.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod
- Restart this workspace and check that terminals are not gone.
- Start a prebuild workspace and check task output: http://akosyakov-verbose-output-from-gitpod-2086.staging.gitpod-dev.com/#prebuild/https://github.com/gitpod-io/gitpod
  - regular to test create page: http://akosyakov-verbose-output-from-gitpod-2086.staging.gitpod-dev.com/#https://github.com/gitpod-io/gitpod
- Check that both terminals history is proper after prebuilt is done.
- Run a prebuild on the repo without init tasks: http://akosyakov-verbose-output-from-gitpod-2086.staging.gitpod-dev.com/#prebuild/https://github.com/jankeromnes/hugo-gitpod
  - regular to test create page: http://akosyakov-verbose-output-from-gitpod-2086.staging.gitpod-dev.com/#https://github.com/jankeromnes/hugo-gitpod
- Run a prebuild on the repo with failing init task: http://akosyakov-verbose-output-from-gitpod-2086.staging.gitpod-dev.com/#prebuild/https://github.com/akosyakov/test-maven-repo/tree/failed_prebuild
  - regular to test create page: http://akosyakov-verbose-output-from-gitpod-2086.staging.gitpod-dev.com/#https://github.com/akosyakov/test-maven-repo/tree/failed_prebuild
